### PR TITLE
Better Web Search Query

### PR DESCRIPTION
**Pull request by KingNish24**
I updated the AI's prompt to yield better search results, understand user demands, and generate useful queries.

Why did I update this?
Because while the current model excels at generating queries for simple and brief tasks, it struggles with complex tasks and unfamiliar terms, often failing to produce appropriate queries.

### CODE REVIEW

**Comment by nsarrazin** [comment]
Hi! Thanks for the contribution, would it be possible to do a comparison table between this query and the old one, to see where it improves things ?

File: src/lib/server/websearch/search/generateQuery.ts [hunk 1]
```diff
@@ -64,9 +64,9 @@ Current Question: Where is it being hosted?`,
 
 	const webQuery = await generateFromDefaultEndpoint({
 		messages: convQuery,
-		preprompt: `You are tasked with generating web search queries. Give me an appropriate query to answer my question for google search. Answer with only the query. Today is ${currentDate}`,
+		preprompt: `You are an expert in crafting concise and effective web search queries. However, keep in that sometimes you may encounter completely unfamiliar terms or concepts. When faced with such uncertainty, please refrain from attempting to substitute or interpret the term in any way. Instead, prioritize maintaining the integrity of the original phrasing as much as possible. Please only provide the query, without answering the question directly.  Your task is to give me the best possible query to answer the user's question on Google Search. Today's date is ${currentDate}`,
```
Comment (gary149) [inline src/lib/server/websearch/search/generateQuery.ts:67]: Any other chat app using this kind of prompt?

**Review by gary149 — APPROVED** [review]

File: src/lib/server/websearch/search/generateQuery.ts (hunk 1 above)
Comment (KingNish24) [reply src/lib/server/websearch/search/generateQuery.ts:67]: > Any other chat app using this kind of prompt?

I don't know. I made it my own tested it.

**Comment by nsarrazin** [comment]
Gonna close this for now, if you have time to provide a table that shows an improvement with the prompt in this PR vs the current one in main let me know and I'll reopen :rocket:

**Pull request closed without merging**